cozy/ui/chapter_element.py

```diff
@@ -17,62 +16,36 @@ class ChapterElement(Gtk.EventBox):
 
     def __init__(self, chapter: Chapter):
         self.selected = False
-        self.chapter: Chapter = chapter
+        self.chapter = chapter
 
         super().__init__()
 
-        self.connect("enter-notify-event", self._on_enter_notify)
-        self.connect("leave-notify-event", self._on_leave_notify)
-        self.connect("button-press-event", self._on_button_press)
-        self.set_tooltip_text(_("Play this part"))
-
-        # This box contains all content
-        self.box = Gtk.Box()
-        self.box.set_orientation(Gtk.Orientation.HORIZONTAL)
-        self.box.set_spacing(3)
-        self.box.set_halign(Gtk.Align.FILL)
-        self.box.set_valign(Gtk.Align.CENTER)
-
-        # These are the widgets that contain data
-        self.play_img = Gtk.Image()
-        no_label = Gtk.Label()
-        title_label = Gtk.Label()
-        dur_label = Gtk.Label()
-
-        self.play_img.set_margin_right(5)
-        self.play_img.props.width_request = 16
-
-        if self.chapter.number > 0:
-            no_label.set_text(str(self.chapter.number))
-        no_label.props.margin = 4
-        no_label.set_margin_right(7)
-        no_label.set_margin_left(0)
-        no_label.set_size_request(30, -1)
-        no_label.set_xalign(1)
+        gesture = Gtk.GestureClick(button=Gdk.BUTTON_PRIMARY)
+        gesture.connect("released", self._on_button_press)
+        self.add_controller(gesture)
+
+        motion = Gtk.EventControllerMotion()
+        motion.connect("enter", self._on_enter_notify)
+        motion.connect("leave", self._on_leave_notify)
+        self.add_controller(motion)
 
         self.number_label.set_text(str(self.chapter.number))
         self.title_label.set_text(self.chapter.name)
         self.duration_label.set_text(seconds_to_str(self.chapter.length))
 
-        self.icon_event_box.connect("enter-notify-event", self._on_enter_notify)
-        self.icon_event_box.connect("leave-notify-event", self._on_leave_notify)
-
-    def _on_button_press(self, _, event):
-        if event.type == Gdk.EventType.BUTTON_PRESS and event.button != 1:
-            return False
-
+    def _on_button_press(self, *_):
         self.emit("play-pause-clicked", self.chapter)
-        return True
```

Comment:
I'm not sure if it's necessary in gtk 4 but in gtk 3 it's necessary to tell gtk that we handled the click. Otherwise it will be propagated to the underlaying widget.

Reply:
Yeah, that's why I wondered. Because with `Gtk.Gesture`s we use `gesture.set_state(Gtk.EventSequenceState.CLAIMED)` to achieve the same. Probably it doesn't really matter in this case though.